packages/core/src/external/carequality/dr/get-document-download-url.ts

```diff
@@ -12,34 +12,41 @@ import { XDSRegistryError } from "../error";
 const region = Config.getAWSRegion();
 const medicalDocumentsBucketName = Config.getMedicalDocumentsBucketName();
 
-export async function getDocumentDownloadURL(
+export async function buildDocumentReferences(
   payload: InboundDocumentRetrievalReq
 ): Promise<DocumentReference[]> {
   validateBasePayload(payload);
 
   const documentIds = extractDocumentIds(payload);
   if (documentIds.length === 0) {
-    throw new XDSRegistryError("Valid Dcument ID is not defined");
+    throw new XDSRegistryError("Valid Document ID is not defined");
   }
 
-  return await retrievePreSignedUrls(documentIds);
+  return await retrieveDocumentReferences(documentIds);
 }
 
-async function retrievePreSignedUrls(documentIds: string[]): Promise<DocumentReference[]> {
+async function retrieveDocumentReferences(documentIds: string[]): Promise<DocumentReference[]> {
   const s3Utils = new S3Utils(region);
   const documentReferences: DocumentReference[] = [];
 
   // TODO consider making this more robust, so if one fails we still return the rest
   for (const id of documentIds) {
-    const url = await s3Utils.getSignedUrl({
-      bucketName: medicalDocumentsBucketName,
-      fileName: id,
-    });
+    const { size, contentType, eTag } = await s3Utils.getFileInfoFromS3(
+      id,
+      medicalDocumentsBucketName
+    );
+    if (!eTag) {
+      const message = `Failed to retrieve ETag for document`;
+      console.log(`${message}: ${id}`);
+      throw new XDSRegistryError("Document Hash is not defined");
+    }
```

Comment:
ditto: is it really required? In this case I think it is, right?

If we keep it around, the message is not very helpful for clients, IMO. They don't know that the hash is used to define the doc ID in this scenario. Maybe something like `Failed to document ID` or something similar?

Reply:
If there is no eTag something is wrong and we should error. Changing error message to "Failed to retrieve Document"